test_conformance/conversions/conversions_data_info.h

```diff
@@ -608,18 +661,30 @@ void DataInfoSpec<InType, OutType>::conv_sat(OutType *out, InType *in)
     }
 }
 
-template <typename InType, typename OutType>
-void DataInfoSpec<InType, OutType>::init(const cl_uint &job_id,
-                                         const cl_uint &thread_id)
+template <typename InType, typename OutType, bool InFP, bool OutFP>
+void DataInfoSpec<InType, OutType, InFP, OutFP>::init(const cl_uint &job_id,
+                                                      const cl_uint &thread_id)
 {
     uint64_t ulStart = start;
     void *pIn = (char *)gIn + job_id * size * gTypeSizes[inType];
 
-    if (std::is_integral<InType>::value)
+    if (is_in_half())
+    {
+        const unsigned sclamp =
+            std::is_signed<OutType>::value ? 0xffff : 0x7fff;
+
+        InType *o = (InType *)pIn;
+
+        for (unsigned i = start; i < size; i++)
+        {
+            o[i] = static_cast<InType>(i % sclamp);
+        }
+    }
+    else if (std::is_integral<InType>::value)
     {
         InType *o = (InType *)pIn;
         if (sizeof(InType) <= sizeof(cl_short))
-        { // char/uchar/ushort/short
+        { // char/uchar/ushort/short/half
```

Comment:
Minor: I think this comment is incorrect - halfs are handled above.

```suggestion
        { // char/uchar/ushort/short
```

Reply:
Done